packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-suggestions.tsx

```diff
@@ -0,0 +1,258 @@
+          part={
+            relatedQuestion.expanded
+              ? 'question-expanded-text'
+              : 'question-collapsed-text'
+          }
+        >
+          {relatedQuestion.question}
+        </Heading>
+      </Button>
+    );
+  }
+
+  private renderContent(relatedQuestion: SmartSnippetRelatedQuestion) {
+    return (
+      <atomic-smart-snippet-answer
+        exportparts="answer"
+        htmlContent={relatedQuestion.answer}
+        innerStyle={this.style}
+      ></atomic-smart-snippet-answer>
+    );
+  }
+
+  private renderSource(relatedQuestion: SmartSnippetRelatedQuestion) {
+    const {source} = relatedQuestion;
+    if (!source) {
+      return [];
+    }
+    const interactiveResult = buildInteractiveResult(this.bindings.engine!, {
+      options: {result: source},
+    });
+    return (
+      <section aria-label={this.bindings.i18n.t('smart-snippet-source')}>
+        <LinkWithResultAnalytics
+          title={source.clickUri}
+          href={source.clickUri}
+          target="_self"
+          part="source-url"
+          onSelect={() => interactiveResult.select()}
+          onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
+          onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
+        >
+          {source.clickUri}
+        </LinkWithResultAnalytics>
+        <LinkWithResultAnalytics
+          title={source.title}
+          href={source.clickUri}
+          target="_self"
+          part="source-title"
+          onSelect={() => interactiveResult.select()}
+          onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
+          onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
+        >
+          {source.title}
+        </LinkWithResultAnalytics>
+      </section>
+    );
+  }
+
+  public renderRelatedQuestion(relatedQuestion: SmartSnippetRelatedQuestion) {
+    return (
+      <li
+        key={this.getRelatedQuestionId(relatedQuestion)}
+        part={
+          relatedQuestion.expanded
+            ? 'question-answer-expanded'
+            : 'question-answer-collapsed'
+        }
+        class="flex flex-col"
+      >
+        {this.renderQuestion(relatedQuestion)}
+        <div
+          part={relatedQuestion.expanded ? 'answer-and-source' : ''}
+          class={relatedQuestion.expanded ? 'pl-10 pr-6 pb-6' : 'hidden'}
+          id={this.getRelatedQuestionId(relatedQuestion)}
+        >
+          {this.renderContent(relatedQuestion)}
+          <footer part="footer">{this.renderSource(relatedQuestion)}</footer>
+        </div>
+      </li>
+    );
+  }
+
+  public render() {
+    if (!this.smartSnippetQuestionsListState.questions.length) {
+      return <Hidden></Hidden>;
+    }
+
+    return (
+      <aside
+        part="container"
+        class="bg-background border border-neutral rounded-lg text-on-background mb-6 overflow-hidden"
+      >
+        <Heading
+          level={this.headingLevel ?? 0}
```

Comment:
if the default is 0, isn't this check necessary?

Reply:
It is, you're right. I'll remove it.